clusterloader2/testing/l4ilb/config.yaml

```diff
@@ -0,0 +1,120 @@
+#Constants
+{{$LARGE_BACKEND_SIZE := 300}}
+{{$MEDIUM_BACKEND_SIZE := 150}}
+{{$SMALL_BACKEND_SIZE := 10}}
+# adding a fixed value for first version of the test, rate of pod creation not a concern yet.
+{{$ilbQPS := 500}}
+{{$namespaces := 1}}
+#Test
+name: ilbload
+automanagedNamespaces: {{$namespaces}}
+tuningSets:
+- name: ILBConstantQPS
+  qpsLoad:
+    averageQps: {{$ilbQPS}}
+steps:
+# Create ILBs
+- measurements:
+  - Identifier: ServiceCreationLatency
+    Method: ServiceCreationLatency
+    Params:
+      action: start
+  - Identifier: WaitForRunningDeployments
+    Method: WaitForControlledPodsRunning
+    Params:
+      action: start
+      apiVersion: apps/v1
+      kind: Deployment
+      operationTimeout: 15m
+
+- phases:
+  - namespaceRange:
+      min: 1
+      max: {{$namespaces}}
+    replicasPerNamespace: 1
+    tuningSet: ILBConstantQPS
+    objectBundle:
+    - basename: large-backends-dep
+      objectTemplatePath: dep.yaml
+      templateFillMap:
+        ReplicasMin: {{$LARGE_BACKEND_SIZE}}
+        ReplicasMax: {{$LARGE_BACKEND_SIZE}}
+    - basename: large-backends-service
+      objectTemplatePath: service.yaml
+      templateFillMap:
+        DeploymentBaseName: large-backends-dep
+- phases:
+  - namespaceRange:
+      min: 1
+      max: {{$namespaces}}
+    replicasPerNamespace: 1
+    tuningSet: ILBConstantQPS
+    objectBundle:
+    - basename: medium-backends-dep
+      objectTemplatePath: dep.yaml
+      templateFillMap:
+        ReplicasMin: {{$MEDIUM_BACKEND_SIZE}}
+        ReplicasMax: {{$MEDIUM_BACKEND_SIZE}}
```

Comment:
I'm wondering what is the point of testing large/medium/small deployments for ILB.
The ILB is build on top of nodes, and based on size of deployments we're only programming iptables, right?

I'm wondering why having just large deployment wouldn't be enough?
It would visibly simplify the test.

Reply:
This is currently the case, we add all nodes to the cluster and program iptables to send the packet to the right deployment pods. However, with ILB subsetting feature, we will create a different backend for each ILB proportional to the size of the deployment. Then this test will be useful to ensure that we pick only nodes where we have a matching endpoint. 

not sure if that can be added as a predefined step:
we would need to list all the matching pods - the nodes they are on and then compare that list with the backend of a given ILB.

Reply:
Re sizes: fair enough

Re predefined step - it's is super provider specific to list backends of ILB. We can probably add such measurement, if we make it very specific that it's for gcp only.
Though I wouldn't block this PR on it.

Reply:
yes, i agree with you on both points.